Marlin/src/feature/power.cpp

```diff
@@ -85,18 +87,23 @@ bool Power::is_power_needed() {
     #endif
   ) return true;
 
-  HOTEND_LOOP() if (thermalManager.degTargetHotend(e) > 0 || thermalManager.temp_hotend[e].soft_pwm_amount > 0) return true;
-  if (TERN0(HAS_HEATED_BED, thermalManager.degTargetBed() > 0 || thermalManager.temp_bed.soft_pwm_amount > 0)) return true;
+  #if HAS_HOTEND
+    HOTEND_LOOP() if (thermalManager.degTargetHotend(e) > 0 || thermalManager.temp_hotend[e].soft_pwm_amount > 0) return true;
+  #endif
+
+  #if HAS_HEATED_BED
+  if (thermalManager.degTargetBed() > 0 || thermalManager.temp_bed.soft_pwm_amount > 0) return true;
+  #endif
 
-  #if HAS_HOTEND && AUTO_POWER_E_TEMP
+  #if HAS_HOTEND && ENABLED(AUTO_POWER_E_TEMP)
     HOTEND_LOOP() if (thermalManager.degHotend(e) >= (AUTO_POWER_E_TEMP)) return true;
   #endif
 
-  #if HAS_HEATED_CHAMBER && AUTO_POWER_CHAMBER_TEMP
+  #if HAS_HEATED_CHAMBER && ENABLED(AUTO_POWER_CHAMBER_TEMP)
```

Comment:
The `ENABLED()` macro cannot be used with options that take a numeric value. So, what we want really is `HAS_HEATED_CHAMBER && AUTO_POWER_CHAMBER_TEMP` which is equivalent to `HAS_HEATED_CHAMBER && AUTO_POWER_CHAMBER_TEMP != 0`.

Reply:
Ahh, I wasn't looking hard enough at `AUTO_POWER_CHAMBER_TEMP` and I think was assuming it was boolean defined/undefined like `AUTO_POWER_COOLER_FAN`. For those, my rationale was that the leftover `if(0) return true` would never be ran, though I suppose that may just be dropped when compiling.

Thank you!